controlplane/src/core/repositories/OrganizationRepository.ts

```diff
@@ -1263,4 +1274,135 @@ export class OrganizationRepository {
+      max: 1,
+      search: org.slug,
+      realm: input.keycloakRealm,
+      briefRepresentation: false,
+    });
+    const adminChildGroup = await input.keycloakClient.fetchChildGroup({
+      realm: input.keycloakRealm,
+      orgSlug: org.slug,
+      kcGroupId: organizationGroups[0].id!,
+      childGroupType: 'admin',
+    });
+    const devChildGroup = await input.keycloakClient.fetchChildGroup({
+      realm: input.keycloakRealm,
+      orgSlug: org.slug,
+      kcGroupId: organizationGroups[0].id!,
+      childGroupType: 'developer',
+    });
+    const viewerChildGroup = await input.keycloakClient.fetchChildGroup({
+      realm: input.keycloakRealm,
+      orgSlug: org.slug,
+      kcGroupId: organizationGroups[0].id!,
+      childGroupType: 'viewer',
+    });
+
+    const allMembers = await this.getMembers({ organizationID: input.organizationId });
+    for (const member of allMembers) {
+      await input.keycloakClient.client.users.delFromGroup({
+        id: member.userID!,
+        realm: input.keycloakRealm,
+        groupId: adminChildGroup.id!,
+      });
+
+      await input.keycloakClient.client.users.delFromGroup({
+        id: member.userID!,
+        realm: input.keycloakRealm,
+        groupId: devChildGroup.id!,
+      });
+
+      await input.keycloakClient.client.users.addToGroup({
+        id: member.userID,
+        realm: input.keycloakRealm,
+        groupId: viewerChildGroup.id!,
+      });
+    }
+
+    await this.db.transaction(async (tx) => {
+      const orgRepo = new OrganizationRepository(this.logger, tx, this.defaultBillingPlanId);
+      const oidcRepo = new OidcRepository(tx);
+
+      await oidcRepo.deleteOidcProvider({ organizationId: input.organizationId });
+
+      for (const member of allMembers) {
+        const userRoles = await orgRepo.getOrganizationMemberRoles({
+          userID: member.userID,
+          organizationID: input.organizationId,
+        });
+        const highPriorityRole = getHighestPriorityRole({ userRoles });
+
+        await orgRepo.updateUserRole({
+          organizationID: input.organizationId,
+          orgMemberID: member.orgMemberID,
+          role: 'viewer',
+          previousRole: highPriorityRole,
+        });
+      }
+
+      await tx
+        .delete(schema.organizationFeatures)
+        .where(eq(schema.organizationFeatures.organizationId, input.organizationId));
+
+      await tx
+        .update(schema.organizations)
+        .set({
+          isDeactivated: true,
+          deactivationInitiatedAt: new Date(),
+          deactivationReason: input.reason,
+        })
+        .where(eq(schema.organizations.id, input.organizationId));
+    });
+
+    const now = new Date();
+    const oneMonthFromNow = addDays(now, 30);
+    const delay = Number(oneMonthFromNow) - Number(now);
+
+    return input.deleteOrganizationQueue.addJob(
+      {
+        organizationId: input.organizationId,
```

Comment:
You can use the organizationID as job id to avoid duplicates. https://docs.bullmq.io/guide/jobs/job-ids

Reply:
Done. It is set in the class method